### PR TITLE
update Income functionality

### PR DESCRIPTION
**Pull request by chaol12**

### CODE REVIEW

File: src/pages/UpdateIncome.jsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import { useAuth } from "../context/AuthContext";
+
+const UpdateIncome = ({showUpdateIncome,setShowUpdateIncome,setMonthlyIncome}) => {
+
+    const {currentUser} = useAuth();
+    const [newIncome, setNewIncome] = useState("") 
+
+    const addTransactions = (Income) => {
+        if(currentUser){
+            upDdateIncome(currentUser.uid,Income)
```
Comment (aappacode) [inline src/pages/UpdateIncome.jsx:14]: is this line updating the income in the database?

File: src/pages/UpdateIncome.jsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            setMonthlyIncome(Income)
+        }else{
+            alert("something went wrong")
```
Comment (aappacode) [inline src/pages/UpdateIncome.jsx:17]: may need more descriptive alert

File: src/pages/UpdateIncome.jsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    const handleSubmit =() =>{
+        setShowUpdateIncome(false)
+        if(newIncome && parseFloat(newIncome) < 1000000000){
+            const absAmount = Math.abs(parseFloat(newIncome).toFixed(2))
+            addTransactions(absAmount)
+        }else
+        {
+          alert("incorrect Input")
```
Comment (aappacode) [inline src/pages/UpdateIncome.jsx:28]: "The amount you entered was inputted correctly. E.g. ..."

**Review by aappacode — APPROVED** [review]
Based on the demo you showed during our meeting, this code is effective. I added some minor questions/comments if you want to review them before merging!

File: src/pages/UpdateIncome.jsx (hunk 1 above)
Comment (chaol12) [reply src/pages/UpdateIncome.jsx:14]: yeah, the line 14 is the update function

File: src/pages/UpdateIncome.jsx (hunk 2 above)
Comment (chaol12) [reply src/pages/UpdateIncome.jsx:17]: I will change it in next "update budget" pr

**Pull request merged**